### PR TITLE
update logic of `WorkloadDetailEndpoints` to consider hostname as a valid link if there is an `ingress` associated

### LINKED ISSUE #8087: Endpoints point to IP and not hostname, which complicates Ingress
Internal reference: SURE-5591
Reported in 2.6.8

The endpoints listed under Deployment Details point to IP and not hostname and are thus not useable. Suggestion is to remove Endpoints line from Deployment Details. 

Request description:
Remove or move the "Endpoints" line in the Description of a deployment.  The new tabs like Ingresses and Services are super useful, and something like this should probably be found under one of these tabs.  The admin reports that the HTTPS link created when you deploy and ingress is particularly not useful to users since the https endpoint doesn't actually map to their ingress, but to the node IP that their users don't have access to. They said that it would be better to just have the different links in the tabs associated with the appropriate ingress or service.

Right now the Services tab has links to the service endpoint in the Target column but the Ingress has a link to the service dashboard instead of the endpoints created by the ingress. 

Here is their initial report verbatim (the request evolved to moving the endpoints to tabs during the call but this provides some context):

Steps to reproduce:
1. Create a workload running a webserver
2. Add an Ingress to the workload
3. Observe in the Deployment details that there is a "443/HTTPS" link in the "Endpoints" part of the deployment details.
4. Click the "443/HTTPS" link

Expected results:
The browser opens a new tab to `https://ingress-hostname/`

Actual results:
The browser opens a new tab to `https://kubernetes-node-ip/`

Support suspects that the "Endpoints" is supposed to refer to the Service(s) associated with the Deployment. But the URL constructed for the "443/HTTPS" link doesn't make any sense in that context either, as the only Service in this case is a ClusterIP referenced in the Ingress, which would have a cluster-internal IP, not the IP of one of the cluster nodes.

Theory: there is broken code where the "443/HTTPS" link is being constructed, where it's mistakenly using the Ingress controller's IP (which is a DaemonSet using hostPort, so any of the cluster's nodes would show up as IPs). But you can't access an application via Ingress using an IP address, you have to provide a hostname.

In all, the current behavior is very confusing for users, who see a clickable "443/HTTPS" link in their Deployment and think that this is where they can go to access their app, then they get frustrated when they get a 404 from the ingress controller (because they didn't specify a hostname).

It's notable that in the "Ingresses" tab in the Deployment detail view, the "Target" URL is correctly clickable to the application's hostname. The "443/HTTPS" link under "Endpoints" should use that URL, not `https://ip-address`, unless there's no Ingress associated.

Actual behavior:
Endpoints list exists in the Description of a deployment

Expected behavior:
Endpoints should be listed in a tab appropriate to what they link to

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #8087 

### Occurred changes and/or fixed issues

- update logic of `WorkloadDetailEndpoints` to consider hostname as a valid link if there is an `ingress` associated
- formatter code cleanup (unused computed property)
- add unit tests

### Areas or cases that should be tested

- Install repository as `git repo` -> `https://github.com/bashofmann/rodeo` - branch `master` in `apps` - `repositories` on `local` cluster
- in `apps` - `chart` on `local` cluster, find a card for `tetris` app that comes from the repo added on the step before and install it
- on `local` cluster, `workloads` - `deployments`, find the deployment associated with the deployment of `tetris`
- On the details in the top of the page, check that the `endpoints` link points to the hostname `tetris.kube-public.172.18.0.3` as per the changes on this PR 

### Screenshot/Video

**Before**
 
before

**After**

after

### CODE REVIEW

File: shell/components/formatter/WorkloadDetailEndpoints.vue [hunk 1]
```diff
@@ -90,7 +75,7 @@ export default {
         :href="endpoint.link"
         target="_blank"
         rel="nofollow noopener noreferrer"
-      ><span v-if="endpoint.port">{{ endpoint.port }}/</span>{{ endpoint.protocol }}</a>
+      >{{ endpoint.link }}</a>
```
Comment (nwmac) [inline shell/components/formatter/WorkloadDetailEndpoints.vue:78]: This changes the display for the service case as well - do we want to do this?

I'd suggest leaving it as is - can we add a property to the endpoint that is the what should be displayed for the link and set that in the loop - the services can use the original from the template.

**Review by nwmac — CHANGES_REQUESTED** [review]

File: shell/components/formatter/WorkloadDetailEndpoints.vue (hunk 1 above)
Comment (aalves08) [reply shell/components/formatter/WorkloadDetailEndpoints.vue]: I think I interpreted this correctly @nwmac 🤞 https://github.com/rancher/dashboard/pull/10380/commits/9defbd34b1f1863fa746dc12f81079d2d07321ac

PR is ready for re-review

File: shell/components/formatter/WorkloadDetailEndpoints.vue [hunk 2]
```diff
@@ -27,17 +27,27 @@ export default {
...
+
           out.forEach((endpoint) => {
             let protocol = 'http';
 
             if (endpoint.port === 443) {
               protocol = 'https';
             }
 
-            if (endpoint.addresses) {
+            const linkDefaultDisplay = endpoint.port ? `${ endpoint.port }${ endpoint.protocol }` : endpoint.protocol;
```
Comment (nwmac) [inline shell/components/formatter/WorkloadDetailEndpoints.vue:38]: Missing a slash between the port and protocol

**Review by nwmac — CHANGES_REQUESTED** [review]
Minor change

File: shell/components/formatter/WorkloadDetailEndpoints.vue (hunk 2 above)
Comment (aalves08) [reply shell/components/formatter/WorkloadDetailEndpoints.vue]: Thanks for catching that 🙏

**Pull request merged**